### ISSUE TITLE
Mapnik .NET and ICU issue

### ISSUE THREAD

**Issue by rouen-sk** [mapnik/mapnik#3597]
I am using Mapnik (3.0.11) in my .NET code (using C++/CLI wrapper over Mapnik Win SDK, see https://github.com/jbrwn/NET-Mapnik). There is one major issue - text symbolizers never wrap text, because Mapnik cant find/use ICU correctly, since in the log i can see `could not create BreakIterator: U_MISSING_RESOURCE_ERROR`

NET-Mapnik have build task, which copies `/share/icu` directory from mapnik's `lib` into the `bin` directory of the application (so there is `PROJECT_DIR\bin\share\icu\icudt56l.dat` present), and sets the evironment variable in entry point (from https://github.com/jbrwn/NET-Mapnik/blob/master/NETMapnik/mapnik.cpp): 

```
BOOLEAN WINAPI DllMain(IN HINSTANCE hDllHandle,
	IN DWORD     nReason,
	IN LPVOID    Reserved)
{
	switch (nReason)
	{
		case DLL_PROCESS_ATTACH:
			_putenv_s("PROJ_LIB", "share\\proj");
			_putenv_s("ICU_DATA", "share\\icu");
			_putenv_s("GDAL_DATA", "share\\gdal");
			break;
		case DLL_THREAD_ATTACH:
		case DLL_THREAD_DETACH:
		case DLL_PROCESS_DETACH:
			break;
	}
	return TRUE;
}
```
I have tried:
1. to change this `_putenv_s("ICU_DATA", "icu");` (and copy the `icu` directory directly to `bin`) 
2. remove this `_putenv_s` entirely and copy the `icudt56l.dat` file directly into `bin`
3. remove this `_putenv_s` entirely, and set the `ICU_DATA` from managed code in my app before calling Mapnik first time

None of this helps, still got `could not create BreakIterator: U_MISSING_RESOURCE_ERROR`

Do I have any means to know what is problem? Maybe see the paths the Mapnik is trying?

**Comment by artemp** [comment]
@rouen-sk : looks related https://github.com/mapnik/mapnik/issues/3193

**Comment by rouen-sk** [comment]
@artemp I have seen this issue when I tried to google my error, but I am not sure what to take from it. From the code sample in node-mapnik it seems like they are setting `PATH`, not `ICU_DATA`, but ICU docs mention only `ICU_DATA` environment variable... Do you suggest I try to add `icu` directory into `PATH` before calling Mapnik?

**Comment by artemp** [comment]
@rouen-sk - I don't have access to windows so I can't comment but why don't you try setting up `PATH` and see if this works. And report back with the status.
/cc @BergWerkGIS

**Comment by rouen-sk** [comment]
Just tried to add absolute path to `icu` directory to PATH before calling Mapnik. Did not help, same error.

**Comment by rouen-sk** [comment]
I just tried another thing: I added the `icu` directory to `PATH` environment variable in OS (not just in process). Now the errors are gone (and I can see my application exe actually using the ICU .dat file via Process Explorer) - but texts in the resulting png are not wrapped. No other errors in mapnik log, so I dont know where to go from here.

**Comment by BergWerkGIS** [comment]
> Just tried to add absolute path to icu directory to PATH before calling Mapnik. Did not help, same error.

@rouen-sk 
In https://github.com/jbrwn/NET-Mapnik/issues/38 you mentioned running mapnik/NET-Mapnik under IIS.
Did you restart IIS after adding ICU directory to PATH?
I don't think IIS will pick it up without restarting.

I think the problem lies that you are running in the IIS process and its working directory.

**Comment by rouen-sk** [comment]
@BergWerkGIS I dont think there is any connection with IIS. I also have console application for testing, and the error was visible there also. But see my last comment - after I added path to `icu` directory into the OS-wide environment variable, the error disappeared - seeems like ICU .dat file is finally found. But texts in generated images are still not wrapped (with no error to point me to reason).

**Comment by BergWerkGIS** [comment]
@rouen-sk oops, wrong thread.
Sorry, no idea why there is no wrapping.

**Comment by rouen-sk** [comment]
It seems like the issue is resolved in mapnik 3.0.13-rc1. I just created wapper for this version and the word wrapping works (without any additional work from me).

**Comment by artemp** [comment]
@rouen-sk - great, for thanks for feedback. Closing.
